modules/p2/ding_list/include/form.inc

```diff
@@ -741,3 +741,52 @@ function ding_list_list_permissions_form_share_callback($form, &$form_state) {
 
   return $response;
 }
+
+/**
+ * Ask user to reset their lists.
+ */
+function ding_list_reset_user_form($form, &$form_state, $account) {
+  $form['warning'] = array(
+    '#markup' => '<p>' . t('Are you sure you want to reset all your lists.') . '</p>',
+  );
+
+  $form['actions'] = array(
+    '#type' => 'actions',
+    'yes' => array(
+      '#type' => 'submit',
+      '#value' => t('Yes'),
+    ),
+    'cancel' => array(
+      '#type' => 'submit',
+      '#value' => t('Cancel'),
+      '#submit' => array('ding_list_reset_user_form_submit_cancel'),
+      '#limit_validation_errors' => array(),
+    ),
+  );
+
+  return $form;
```

Comment:
This form should be replaced by [`confirm_form`](https://api.drupal.org/api/drupal/modules%21system%21system.module/function/confirm_form/7.x) unless there is a specific reason to implement this ourselves.